dbt/exceptions.py

```diff
@@ -101,8 +102,62 @@ def type(self):
         return 'Compilation'
 
 
+EXTRA_PROPERTIES = re.compile(
+    r'Additional properties are not allowed \(([^)]*) (were|was) unexpected\)'
+)
+MISSING_PROPERTIES = re.compile(r'(.*) is a required property')
```

Comment:
are these messages returned by the json schema validator? is there an api where we can get structured data from the validation instead of strings?

Reply:
Sort of. You can get that there are extra properties or that there are missing properties, but the names of those properties are only exposed in the message, which is why I went through all these ugly machinations.